### PR TITLE
typescript: update resources/util.ts

### PR DESCRIPTION
**Pull request by quisquous**
This also updates the mocha options to support running with typescript.
On the downside, this also makes tests go as slow as they used to
(~30s) before the recent speedup.

### CODE REVIEW

**Comment by quisquous** [comment]
I don't really know what I'm doing with typescript yet.  I assume I should move Job and Role to live in a definition file?

**Comment by quisquous** [comment]
cc @MaikoTan if you have any time to look at this, since I think this will fix some of the test issues from your patch

File: resources/util.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    canStun: (job: Job) => stunJobs.includes(job),
+    canSilence: (job: Job) => silenceJobs.includes(job),
+    canSleep: (job: Job) => sleepJobs.includes(job),
+    canCleanse: (job: Job) => cleanseJobs.includes(job),
+    canFeint: (job: Job) => feintJobs.includes(job),
+    canAddle: (job: Job) => addleJobs.includes(job),
+  };
+})();
+
+export default Util;
```
Comment (Trim21) [inline resources/util.ts:152]: Why IIFE?

Comment (quisquous) [reply resources/util.ts:152]: Mmmmm I can't remember why it was that way originally.  I'll remove the IIFE to make it cleaner.

**Pull request merged**